code/game/objects/items/weapons/pneumaticCannon.dm

```diff
@@ -28,118 +38,120 @@
 			return
-		updateTank(W, 0, user)
+		add_tank(W, user)
 		return
 	if(W.type == type)
 		to_chat(user, "<span class='warning'>You're fairly certain that putting a pneumatic cannon inside another pneumatic cannon would cause a spacetime disruption.</span>")
 		return
-	if(istype(W, /obj/item/wrench))
-		switch(pressureSetting)
-			if(1)
-				pressureSetting = 2
-			if(2)
-				pressureSetting = 3
-			if(3)
-				pressureSetting = 1
-		to_chat(user, "<span class='notice'>You tweak \the [src]'s pressure output to [pressureSetting].</span>")
-		return
-	if(loadedWeightClass >= maxWeightClass)
-		to_chat(user, "<span class='warning'>\The [src] can't hold any more items!</span>")
-		return
-	if(isitem(W))
-		var/obj/item/IW = W
-		if(IW.flags & (ABSTRACT | NODROP | DROPDEL))
-			to_chat(user, "<span class='warning'>You can't put [IW] into [src]!</span>")
-			return
-		if((loadedWeightClass + IW.w_class) > maxWeightClass)
-			to_chat(user, "<span class='warning'>\The [IW] won't fit into \the [src]!</span>")
-			return
-		if(IW.w_class > src.w_class)
-			to_chat(user, "<span class='warning'>\The [IW] is too large to fit into \the [src]!</span>")
-			return
-		if(!user.unEquip(W))
-			return
-		to_chat(user, "<span class='notice'>You load \the [IW] into \the [src].</span>")
-		loadedItems.Add(IW)
-		loadedWeightClass += IW.w_class
-		IW.loc = src
-		return
+	load_item(W, user)
 
 /obj/item/pneumatic_cannon/screwdriver_act(mob/living/user, obj/item/I)
-	if(!tank)
-		return
-
-	updateTank(tank, 1, user)
+	remove_tank(user)
 	return TRUE
 
 /obj/item/pneumatic_cannon/afterattack(atom/target, mob/living/carbon/human/user, flag, params)
 	if(isstorage(target)) //So you can store it in backpacks
 		return ..()
-	if(istype(target, /obj/structure/closet)) //So you can store it in closets
-		return ..()
 	if(istype(target, /obj/structure/rack)) //So you can store it on racks
 		return ..()
 	if(!istype(user))
 		return ..()
-	Fire(user, target)
+	fire(user, target)
 
 
-/obj/item/pneumatic_cannon/proc/Fire(mob/living/carbon/human/user, atom/target)
+/obj/item/pneumatic_cannon/proc/fire(mob/living/carbon/human/user, atom/target)
 	if(!istype(user) && !target)
 		return
-	var/discharge = 0
-	if(!loadedItems || !loadedWeightClass)
-		to_chat(user, "<span class='warning'>\The [src] has nothing loaded.</span>")
-		return
-	if(!tank)
-		to_chat(user, "<span class='warning'>\The [src] can't fire without a source of gas.</span>")
-		return
-	if(tank && !tank.air_contents.remove(gasPerThrow * pressureSetting))
-		to_chat(user, "<span class='warning'>\The [src] lets out a weak hiss and doesn't react!</span>")
+	var/has_discharged = FALSE
+	if(!loaded_items || !loaded_weight_class)
+		to_chat(user, "<span class='warning'>[src] has nothing loaded.</span>")
 		return
+	if(requires_tank)
+		if(!tank)
+			to_chat(user, "<span class='warning'>[src] can't fire without a source of gas.</span>")
+			return
+		if(!tank.air_contents.remove(gas_per_throw * pressure_setting))
```

Comment:
Since you're refactoring it, this should use `boolean_remove(gas_per_throw * pressure_setting)` to prevent the cannon from firing when gas is present but less than the intended removal amount.